src/components/SingleReservation.js

```diff
@@ -0,0 +1,38 @@
+import React, { Component } from 'react'
+
+export default class SingleResrvation extends Component {
+
+    updateStatus=()=>{
+        var condition
+        if (this.props.single.status=== "free") {
+            condition= "cancel"
+            this.props.update(this.props.single,condition )
+        }else if (this.props.single.status === "cancel") {
+            condition= "free"
+            this.props.update(this.props.single,condition )
```

Comment:
if (this.props.single.status=== "free") {
            condition = "cancel"
} else if (this.props.single.status === "cancel") {
            condition= "free"
}
this.props.update(this.props.single,condition)

Reply:
You can do like this instead of writing twice this.props.update